Recon/src/Recon/OverlayProducer.cxx

```diff
@@ -358,26 +360,26 @@ void OverlayProducer::onProcessStart() {
   }
 
   // replace by this line once the corresponding tweak to EventFile is ready:
-  //	overlayFile_ = std::make_unique<framework::EventFile>( overlayFileName_,
-  //true );
-  overlayFile_ = std::make_unique<framework::EventFile>(overlayFileName_);
+  overlayFile_ = std::make_unique<framework::EventFile>( overlayFileName_,true );
   overlayFile_->setupEvent(&overlayEvent_);
-
   // we update the iterator at the end of each event. so do this once here to
   // grab the first event in the processor
-  // TODO this could also be done N random times to get a randomness in which
-  // events get matched to what sim event. noticed that shifting by a fair chunk
-  // helps remove some weak but suspicious correlations between sim and overlay
-  // particle positions. leave it hardwired until we can reset the overlay event
-  // counter in nextEvent() (implemented with the EventFile definition above)
-  int nEventsShift_ = 23;
-  for (int iShift = 0; iShift < nEventsShift_; iShift++) {
+
+  TRandom2 * rd = new TRandom2( 0.); //RNSS undefined before produce()
+  // shift away a bit from the first event to avoid some weak correlations
+  // seen during initial valiadations
+  // TODO use the actual number of events in the file as upper limit
+  int startEvent = rd->Uniform(20., 500.); //assume we have at least 500 events
+  // and if not, event number wrapping will take care of it
+  for (int iShift = 0; iShift < startEvent; iShift++) {
     if (!overlayFile_->nextEvent()) {
       std::cerr << "Couldn't read next event!";
       return;
     }
   }
```

Comment:
Now that I think about it... Since you are already making some small updates to the EventFile class, you could just add a `skip` method. In that method, you would have access to the number of events in the file, so OverlayProducer gives a randomly generated offset to the skip method and then...

```cpp
bool EventFile::skip(int off) {
  off = off % entries_; // don't want to waste time looping
  for (int i_shift{0}; i_shift < off; i_shift++) {
    if (!this->nextEvent()) return false;
  }
  return true;
}
```

Then your initialization procedure here would just need to be
```cpp
int start_event = rd->Uniform(0.,1e6); // EventFile::skip handles number of events in file
if (!overlayFile_->skip(start_event)) {
  EXCEPTION_RAISE("BadRead",
    "Couldn't read to starting offset.");
}
```

Reply:
True, but in that case, why loop at all? Since we have already established that we're within the actual number of events, can't I just implement a setter for the entry number instead? Or a better name like  
```
bool EventFile::skipToEvent(int off) {
  ientry_ = off % entries_; // don't want to waste time looping
  if (!this->nextEvent()) return false;
  return true;
} 
```

Reply:
Implemented it without the loop, works.